demos/fleet_provisioning/fleet_provisioning_with_csr/demo_config.h

```diff
@@ -158,7 +158,9 @@
  *
  * This is passed to MbedTLS; see https://tls.mbed.org/api/x509__csr_8h.html#a954eae166b125cea2115b7db8c896e90
  */
-#define SUBJECT_NAME    "CN=Fleet Provisioning Demo"
```

Comment:
Can you update the CI build configuration as well as update the following CMake configuration to allow `CSR_SUBJECT_NAME` to be passed as a CMake variable?
https://github.com/aws/aws-iot-device-sdk-embedded-C/blob/main/demos/fleet_provisioning/fleet_provisioning_with_csr/CMakeLists.txt#L59-L70